healthcare/healthcare/doctype/patient_appointment/patient_appointment.js

```diff
@@ -529,17 +529,35 @@ let check_and_set_availability = function(frm) {
 					count_class = `${(available_slots > 0 ? 'badge-success' : 'badge-danger')}`;
 					tool_tip =`${available_slots} ${__('slots available for booking')}`;
 				}
-				return `
+				// restrict past slots based on the current time.
+				let now = moment();
+				if(slot_start_time.isBefore(now) && (now.format("YYYY-MM-DD") == current_date)){
+					return `
```

Comment:
It would be better instead of returning another button for disabled, just pass "disabled = true" based on your condition, then single button is enough, (for reference this is done above for already booked slots).